tests/test_exporter_git.py

```diff
@@ -0,0 +1,206 @@
+)
+def test_export_model_happy(
+    t4c_exporter_container: containers.Container, t4c_ip_addr: str
+):
+    conftest.wait_for_container(
+        t4c_exporter_container,
+        "Export of model to TeamForCapella server finished",
+    )
+
+    t4c_projects: list[
+        dict[str, str]
+    ] = conftest.get_projects_of_t4c_repository(t4c_ip_addr)
+
+    assert len(t4c_projects) == 1
+
+    t4c_project: dict[str, str] = t4c_projects[0]
+
+    assert t4c_project["repositoryName"] == "test-repo"
+    assert t4c_project["projectName"] == "test-project"
+    assert t4c_project["path"] == "test-project"
+
+
+@pytest.mark.skipif(
+    condition=conftest.is_capella_5_x_x(),
+    reason="Exporter does not work for capella 5.x.x",
+)
+@pytest.mark.parametrize(
+    "t4c_exporter_git_env",
+    [
+        pytest.param(
+            {"ENTRYPOINT": "/this/does/not/exist"},
+            id="invalid-entrypoint",
+        ),
+        pytest.param(
+            {"GIT_REPO_BRANCH": "unknown-branch"}, id="unkown-branch"
+        ),
+        pytest.param({"T4C_REPO_HOST": "unknown"}, id="unknown-repo-host"),
+        pytest.param({"T4C_REPO_NAME": "unkown"}, id="unknown-repo-name"),
+    ],
+    indirect=True,
+)
+def test_export_model_6_x_x_unhappy(
+    t4c_exporter_container: containers.Container,
+):
+    with pytest.raises(RuntimeError):
+        conftest.wait_for_container(t4c_exporter_container, "Export finished")
+
+
+@pytest.mark.skipif(
+    condition=conftest.is_capella_6_x_x(),
+    reason="Tests checkes whether error is thrown for 5.x.x",
+)
+def test_export_model_5_x_x_unhappy(
+    t4c_exporter_container: containers.Containe,
+):
+    with pytest.raises(RuntimeError):
+        conftest.wait_for_container(t4c_exporter_container, "Export finished")
+
+
+def _checkout_git_repository(server_ip: str, path: pathlib.Path):
+    subprocess.run(  # pylint: disable=subprocess-run-check
+        [
+            "git",
+            "clone",
+            f"http://{server_ip}:80/git/git-test-repo.git",
+            path,
+        ],
+        **SUBPROCESS_DEFAULT_ARGS,
+    )
+    subprocess.run(  # pylint: disable=subprocess-run-check
+        ["git", "switch", "-c", conftest.GIT_REPO_BRANCH],
+        cwd=path,
+        **SUBPROCESS_DEFAULT_ARGS,
+    )
+
+
+def _copy_test_project_into_git_repo(git_path: pathlib.Path):
+    target_dir: pathlib.Path = pathlib.Path(
+        git_path,
+        str(pathlib.Path(ENTRYPOINT).parent).lstrip("/"),
+    )
+    target_dir.mkdir(exist_ok=True, parents=True)
+
+    project_dir: pathlib.Path = (
+        pathlib.Path(__file__).parents[0] / "data" / "test-project"
+    )
+
+    for file in project_dir.glob("*"):
+        if not str(file).endswith("license"):
+            shutil.copy2(file, target_dir)
+
+
+def _commit_and_push_git_repo(path: pathlib.Path):
+    subprocess_shared_args: dict[str, t.Any] = SUBPROCESS_DEFAULT_ARGS | {
+        "cwd": path
+    }
+
+    subprocess.run(  # pylint: disable=subprocess-run-check
```

Comment:
Why did you disable `subprcoess-run-check` here?

Reply:
I moved the default subprocess arguments (`check=True, text=True, capture_output=True`) to the `SUBPROCESS_DEFAULT_ARGS` dict but pylint says that the `check=True` is missing.

Reply:
Ouch. Stupid pylint :D